f600233d1b154d507b782f84604bcb12/update/sys_script_include_abc17e0edba4e918b1227ea5f39619fc.xml

```diff
@@ -7,237 +7,568 @@
-						return "Secret";
-					case '05':  //0005 Cloud TOP SECRET FFP
-					case '21':  //0021 CLoud TOP SECRET T&M
-						return "Top Secret";
-					default:
-						return -1;
-				}
-			}).filter(function(level) {
-				return level != -1;
-			}).filter(function(lineItem, index, self) {
-				return self.indexOf(lineItem) === index;
+		}
+		if (!environments || environments.length === 0) {
+			throw new Error(JSON.stringify({
+				code: "0400",
+				message: "No environments in request body"
+			}));				
+		}
+		
+        // Create new Task Order parent record
+        const taskOrderRecord = new GlideRecord("x_g_dis_atat_task_order");
+        taskOrderRecord.initialize();
+        taskOrderRecord.setValue("task_order_number", taskOrder.taskOrderNumber);
+        taskOrderRecord.setValue("pop_start_date", taskOrder.popStartDate);
+        taskOrderRecord.setValue("pop_end_date", taskOrder.popEndDate);
+        const newTaskOrderSysId = taskOrderRecord.insert();	
+
+        // Create CLIN entries
+        let clinSysIds = [];
+		Object.keys(taskOrder.clinData).forEach(number => {
+            clinSysIds.push(this.createCLINGlideRecord(number, taskOrder.clinData[number], newTaskOrderSysId));
+        });
+        
+        // Update Task Order to add newly created CLIN children
+        taskOrderRecord.setValue("clins", clinSysIds.toString());
+        taskOrderRecord.update();
+
+		// Create Portfolio
+        const portfolio = new GlideRecord("x_g_dis_atat_portfolio");
+        portfolio.initialize();
+        portfolio.setValue("name", portfolioName);
+        portfolio.setValue("portfolio_owner", gs.getUserID());
+        
+        // TODO: remove line below adding requesting user also as manager
+        // after servicenow issue on PROD (and ONLY on PROD) fixed where 
+        // users when only owners can't see portfolio
+        portfolio.setValue("portfolio_managers", gs.getUserID());
+        portfolio.setValue("agency", portfolioAgency);
+        
+        // set CSP column with first (?) environment name from POST body
+        const cspName = environments?.[0]?.cspName;
+            
+        const optRecord = new global.GlideQuery('x_g_dis_atat_cloud_service_provider')
+			.where('name', cspName)
+			.selectOne('sys_id');	
+        
+        if(!optRecord.isEmpty()) {
+            const cspRecord = optRecord.get();
+            gs.info(`CSP Record: ${JSON.stringify(cspRecord)}`);
+            portfolio.setValue("csp", cspRecord.sys_id);
+        }
+
+        if (acquisitionPackageRecord.sys_id) {
+            portfolio.setValue("acquisition_package", acquisitionPackageRecord.sys_id);
+        }
+        portfolio.insert();
+
+		const taskOrderRecord2 = new GlideRecord("x_g_dis_atat_task_order");
+		taskOrderRecord2.get(newTaskOrderSysId);
+		taskOrderRecord2.setValue("portfolio", portfolio.sys_id);
+		taskOrderRecord2.update();
+
+		environments.forEach((env) => this.createEnvironment(env,portfolioName,portfolio));
```

Comment:
This actually needs to go below the part where we set the active TO (a few lines down).